extensions/amp-copy/0.1/amp-copy.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {user} from '../../../src/log';
+import {CSS} from '../../../build/amp-copy-0.1.css';
+
+/** @const */
+const IOS_USER_AGENT = /ipad|ipod|iphone/i;
+
+/** @const */
+const COPY_SUCCESS = 'Copied!';
+
+/** @const */
+const IOS_SELECT = 'Tap the selected text to copy.';
+
+/** @const */
+const COPY_ERROR = 'This browser does not support copying. Please select the text manually, and copy';
+
+class AmpCopy extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /**
+     * @private {?Attribute}
+     */
+    this.copyText_ = null;
+
+    /**
+     * @private {?Element}
+     */
+    this.displayedText_ = null;
+
+    /**
+     * @private {?Element}
+     */
+    this.childContainer_ = null;
+
+    /**
+     * @private {?Element}
+     */
+    this.copyBtn_ = null;
+
+    /**
+     * @private {?Element}
+     */
+    this.copyNotification_ = null;
+
+    /**
+     * @private {?Boolean}
+     */
+    this.isIos_ = null;
+
+    /**
+     * @private {?Timeout}
+     */
+    this.currentNotificationTimeout_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    //Get our copy text attribute, and assert its existance
```

Comment:
The idea i would think might work well and is most visually pleasing is 

Have the copy text in a div and let it be custom stylable. in your build callback create a hidden+disabled textarea (same height as that of the div), this canbe hidden behind the div . populate the textarea with div.innerText. use this content for copy pasting?